estela-api/api/views/project.py

```diff
@@ -59,6 +59,29 @@ def get_queryset(self):
             else self.request.user.project_set.filter(deleted=False)
         )
 
+    @swagger_auto_schema(
+        methods=["GET"],
+        manual_parameters=[
+            openapi.Parameter(
+                "search",
+                openapi.IN_QUERY,
+                description="Search for a project by name.",
+                type=openapi.TYPE_STRING,
+                required=False,
+            )
+        ],
+        responses={status.HTTP_200_OK: ProjectSerializer(many=True)},
+    )
+    @action(methods=["GET"], detail=False)
+    def search(self, request, *args, **kwargs):
+        search = request.query_params.get("search", None)
+        projects = self.get_queryset()
+        if search:
+            projects = projects.filter(name__icontains=search)
+
+        serializer = ProjectSerializer(projects, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
This returns all the results. It should paginate the results of the matched entries.

Reply:
Done